undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -404,15 +416,35 @@ public Duration getDefaultSessionTimeout() {
                     }
                 };
                 CapabilityServiceConfigurator factoryConfigurator = provider.getSessionManagerFactoryServiceConfigurator(factoryServiceName, configuration);
-                CapabilityServiceConfigurator codecConfigurator = provider.getSessionIdentifierCodecServiceConfigurator(codecServiceName, configuration);
-
                 sessionManagerFactory = builder.requires(factoryConfigurator.getServiceName());
-                sessionIdentifierCodec = builder.requires(codecConfigurator.getServiceName());
-
                 factoryConfigurator.configure(capabilitySupport).build(serviceTarget).install();
-                codecConfigurator.configure(capabilitySupport).build(serviceTarget).install();
+
+                CookieConfig affinityCookieConfig = servletContainer.getAffinityCookieConfig();
+
+                if (affinityCookieConfig == null) {
+                    CapabilityServiceConfigurator codecConfigurator = provider.getSessionIdentifierCodecServiceConfigurator(codecServiceName, configuration);
+                    sessionIdentifierCodec = builder.requires(codecConfigurator.getServiceName());
+                    sessionConfigWrapperSupplier = new Supplier<>() {
+                        @Override
+                        public SessionConfigWrapper get() {
+                            return new CodecSessionConfigWrapper(sessionIdentifierCodec.get());
+                        }
+                    };
+                    codecConfigurator.configure(capabilitySupport).build(serviceTarget).install();
+                } else {
+                    CapabilityServiceConfigurator affinityConfigurator = provider.getAffinityLocatorServiceConfigurator(affinityServiceName, configuration);
+                    affinityLocator = builder.requires(affinityConfigurator.getServiceName());
+                    sessionConfigWrapperSupplier = new Supplier<>() {
+                        @Override
+                        public SessionConfigWrapper get() {
+                            return new AffinitySessionConfigWrapper(affinityCookieConfig, affinityLocator.get());
+                        }
+                    };
+                    affinityConfigurator.configure(capabilitySupport).build(serviceTarget).install();
+                }
```

Comment:
This logic should also apply to WARs that use a shared session manager.

Reply:
Fixed. This is unfortunate consequence after so many rebases, should be all fixed now.